core/mempool/src/lib.rs

```diff
@@ -137,9 +147,27 @@ impl Pool {
         Ok(())
     }
 
+    pub fn add_payload_with_author(&self, payload: ChainPayload, author: AuthorityId) -> Result<(), String> {
```

Comment:
this function has exactly the same signature (and almost the same logic) as function below - add_payload_snapshot

Reply:
their the expectation is that it's final snapshot of proposal though, but seems reasonable to do the same things for keeping track of tx/receipts/known